pkg/middleware/auth.go

```diff
@@ -47,7 +48,16 @@ func notAuthorized(c *m.ReqContext) {
 		return
 	}
 
-	c.SetCookie("redirect_to", url.QueryEscape(setting.AppSubUrl+c.Req.RequestURI), 0, setting.AppSubUrl+"/", nil, false, true)
+	c.Resp.Header().Del("Set-Cookie")
+	cookie := http.Cookie{
+		Name:     "redirect_to",
+		Value:    url.QueryEscape(setting.AppSubUrl + c.Req.RequestURI),
+		HttpOnly: true,
+		Path:     setting.AppSubUrl + "/",
+		Secure:   setting.CookieSecure,
+		MaxAge:   0,
+	}
+	http.SetCookie(c.Resp, &cookie)
```

Comment:
This part seems like we have in other places. This is also not setting the same_site attribute.

We have https://github.com/grafana/grafana/blob/master/pkg/middleware/middleware.go#L243-L270  which is doing similar things but specific to session cookie - this should be able to use shared code for creating and writing the actual cookie. 

We also have https://github.com/grafana/grafana/blob/fd2131c1e32e7d47aad5d35ad0babdb396d6f3c4/pkg/api/login_oauth.go#L228-L245 which does similar things.

Possible to extract the part that at least creates the cookie, maybe to util package or something?